core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -114,7 +114,7 @@ protected ManifestReader(
       this.spec = readPartitionSpec(file);
     }
 
-    this.fileSchema = new Schema(DataFile.getType(spec.partitionType()).fields());
+    this.fileSchema = new Schema(DataFile.getType(spec.rawPartitionType()).fields());
```

Comment:
I think this is fine, but do we actually need the rawPartitionType here? Shouldn't we load correctly with the reassigned one as well?

Reply:
Yea the biggest problem is, the manifests are written with the original partition id, so you need that back to read the manifest.  After this point, everything is re-assigned.